packages/name-controller/src/NameController.ts

```diff
@@ -176,6 +177,7 @@ export class NameController extends BaseController<
     request: UpdateProposedNamesRequest,
   ): Promise<UpdateProposedNamesResult> {
     this.#validateUpdateProposedNamesRequest(request);
+    this.#cleanupExpiredEntries();
```

Comment:
There is also a possible argument that we don't want to iterate through every name on every proposed name update so we could track this with a new property such as `lastClean` to ensure it only runs every hour for example.

At this stage though, that feels like overkill since it's still likely super fast even if the user had 10,000 names for example.

Reply:
Good call, I've implemented this in https://github.com/MetaMask/core/pull/3748/commits/2d4c932e930726b99530cb5457f3ad598039d865

Reply:
Sorry for the confusion Goktug!

This is a great implementation but my comment was saying that we should not do this yet as there is likely very little benefit given the volumes of petnames we expect to have.

Purely for the sake of maintainability and the simplest possible code, it's probably safest to revert this, but it will remain in the PR so we can always add in future if needed.

Reply:
Ah sorry, it's my misunderstood. 

No worries at all, I just reverted those commits already!